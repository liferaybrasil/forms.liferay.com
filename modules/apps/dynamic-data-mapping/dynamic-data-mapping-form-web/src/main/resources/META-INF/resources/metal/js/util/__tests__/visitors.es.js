import {PagesVisitor} from '../visitors.es';
import mockPages from './__mock__/mockPages.es';

let visitor;

describe(
	'PagesVisitor',
	() => {
		beforeEach(
			() => {
				visitor = new PagesVisitor(mockPages);
			}
		);

		afterEach(
			() => {
				if (visitor) {
					visitor.dispose();
				}
			}
		);

		it(
			'should be able to change pages',
			() => {
				expect(
					visitor.mapPages(
						(page, index) => (
							{
								...page,
								title: `New title ${index}`
							}
						)
					)
				).toMatchSnapshot();
			}
		);

		it(
			'should be able to change rows',
			() => {
				expect(
					visitor.mapRows(
						row => (
							{
								...row,
								columns: []
							}
						)
					)
				).toMatchSnapshot();
			}
		);

		it(
			'should be able to change columns',
			() => {
				expect(
					visitor.mapColumns(
						column => (
							{
								...column,
								size: 6
							}
						)
					)
				).toMatchSnapshot();
			}
		);

		it(
			'should be able to change fields',
			() => {
				expect(
					visitor.mapFields(
						(field, index) => (
							{
								...field,
								label: `New label ${index}`
							}
						)
					)
				).toMatchSnapshot();
			}
		);
	}
);