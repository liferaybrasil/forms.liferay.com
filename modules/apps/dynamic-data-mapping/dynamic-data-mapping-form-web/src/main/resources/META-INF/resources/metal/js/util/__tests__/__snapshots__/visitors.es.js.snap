// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagesVisitor should be able to change columns 1`] = `
Array [
  Object {
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "Radio",
                "type": "radio",
              },
              Object {
                "label": "Text field",
                "type": "text",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 6,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "Text",
                "required": true,
                "type": "text",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [
              Object {
                "items": Array [
                  Object {
                    "name": "Foo",
                  },
                  Object {
                    "name": "Bar",
                  },
                ],
                "label": "Select",
                "required": true,
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 6,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "Second row",
                "required": true,
                "type": "text",
              },
            ],
            "size": 6,
          },
        ],
      },
    ],
    "title": "Untitled name",
  },
]
`;

exports[`PagesVisitor should be able to change fields 1`] = `
Array [
  Object {
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "New label 0",
                "type": "radio",
              },
              Object {
                "label": "New label 1",
                "type": "text",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "New label 0",
                "required": true,
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [
              Object {
                "items": Array [
                  Object {
                    "name": "Foo",
                  },
                  Object {
                    "name": "Bar",
                  },
                ],
                "label": "New label 0",
                "required": true,
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "New label 0",
                "required": true,
                "type": "text",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Untitled name",
  },
]
`;

exports[`PagesVisitor should be able to change pages 1`] = `
Array [
  Object {
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "Radio",
                "type": "radio",
              },
              Object {
                "label": "Text field",
                "type": "text",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "Text",
                "required": true,
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [
              Object {
                "items": Array [
                  Object {
                    "name": "Foo",
                  },
                  Object {
                    "name": "Bar",
                  },
                ],
                "label": "Select",
                "required": true,
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "label": "Second row",
                "required": true,
                "type": "text",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "New title 0",
  },
]
`;

exports[`PagesVisitor should be able to change rows 1`] = `
Array [
  Object {
    "rows": Array [
      Object {
        "columns": Array [],
      },
      Object {
        "columns": Array [],
      },
      Object {
        "columns": Array [],
      },
    ],
    "title": "Untitled name",
  },
]
`;
